src/subdomains/core/history/services/history.service.ts

```diff
@@ -0,0 +1,232 @@
+
+      case ExportType.COMPACT:
+        return CompactHistoryDtoMapper.mapBuyCryptoTransactions(buyCryptos).sort((tx1, tx2) =>
+          tx1.date.getTime() > tx2.date.getTime() ? -1 : 1,
+        ) as HistoryDto<T>[];
+    }
+  }
+
+  private getBuyFiatTransactions<T>(buyFiats: BuyFiat[], exportFormat: T): HistoryDto<T>[] {
+    switch (exportFormat) {
+      case ExportType.COIN_TRACKING:
+        return this.fixDuplicateTxCT(CoinTrackingHistoryDtoMapper.mapBuyFiatTransactions(buyFiats)).sort((tx1, tx2) =>
+          tx1.date.getTime() > tx2.date.getTime() ? -1 : 1,
+        ) as HistoryDto<T>[];
+
+      case ExportType.CHAIN_REPORT:
+        return this.fixDuplicateTxCR(ChainReportHistoryDtoMapper.mapBuyFiatTransactions(buyFiats)).sort((tx1, tx2) =>
+          tx1.timestamp.getTime() > tx2.timestamp.getTime() ? -1 : 1,
+        ) as HistoryDto<T>[];
+
+      case ExportType.COMPACT:
+        return CompactHistoryDtoMapper.mapBuyFiatTransactions(buyFiats).sort((tx1, tx2) =>
+          tx1.date.getTime() > tx2.date.getTime() ? -1 : 1,
+        ) as HistoryDto<T>[];
+    }
+  }
+
+  private getStakingInvests<T>(
+    deposits: CryptoStaking[],
+    withdrawals: CryptoStaking[],
+    exportFormat: T,
+  ): HistoryDto<T>[] {
+    switch (exportFormat) {
+      case ExportType.COIN_TRACKING:
+        return this.fixDuplicateTxCT([
+          ...CoinTrackingHistoryDtoMapper.mapStakingDeposits(deposits),
+          ...CoinTrackingHistoryDtoMapper.mapStakingWithdrawals(withdrawals),
+        ]).sort((tx1, tx2) => (tx1.date.getTime() > tx2.date.getTime() ? -1 : 1)) as HistoryDto<T>[];
+
+      case ExportType.CHAIN_REPORT:
+        return this.fixDuplicateTxCR([
+          ...ChainReportHistoryDtoMapper.mapStakingDeposits(deposits),
+          ...ChainReportHistoryDtoMapper.mapStakingWithdrawals(withdrawals),
+        ]).sort((tx1, tx2) => (tx1.timestamp.getTime() > tx2.timestamp.getTime() ? -1 : 1)) as HistoryDto<T>[];
+
+      case ExportType.COMPACT:
+        return [];
+    }
+  }
+
+  private getStakingRewards<T>(
+    stakingRewards: StakingReward[],
+    stakingRefRewards: StakingRefReward[],
+    exportFormat: T,
+  ): HistoryDto<T>[] {
+    switch (exportFormat) {
+      case ExportType.COIN_TRACKING:
+        return this.fixDuplicateTxCT([
+          ...CoinTrackingHistoryDtoMapper.mapStakingRewards(stakingRewards),
+          ...CoinTrackingHistoryDtoMapper.mapStakingRefRewards(stakingRefRewards),
+        ]).sort((tx1, tx2) => (tx1.date.getTime() > tx2.date.getTime() ? -1 : 1)) as HistoryDto<T>[];
+
+      case ExportType.CHAIN_REPORT:
+        return this.fixDuplicateTxCR([
+          ...ChainReportHistoryDtoMapper.mapStakingRewards(stakingRewards),
+          ...ChainReportHistoryDtoMapper.mapStakingRefRewards(stakingRefRewards),
+        ]).sort((tx1, tx2) => (tx1.timestamp.getTime() > tx2.timestamp.getTime() ? -1 : 1)) as HistoryDto<T>[];
+
+      case ExportType.COMPACT:
+        return [];
+    }
+  }
```

Comment:
We probably still need to support staking transactions for compact export, don't you think?